eventkitd/private/protostream/reader.go

```diff
@@ -0,0 +1,37 @@
+package protostream
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+type Reader struct {
+	base io.Reader
+}
+
+func NewReader(base io.Reader) *Reader {
+	return &Reader{
+		base: base,
+	}
+}
+
+func (r *Reader) Unmarshal(pb proto.Message) error {
+	var header [4]byte
+	_, err := io.ReadFull(r.base, header[:])
+	if err != nil {
+		return err
+	}
+	size := binary.BigEndian.Uint32(header[:])
+	if size > maxSerializedSize {
+		return fmt.Errorf("frame size larger than max")
+	}
+	buf := make([]byte, size)
```

Comment:
Note, this can avoid allocation, by using [maxSerializedSize]byte

Reply:
is using 4mb of stack really better than an allocation? okay

Reply:
no, i think 4MB of stack is worse than a small allocation.

Reply:
Ah, indeed. For some reason I was thinking it was 9KB or something.